File: dcp-solutions/2024-12-09/solution.go
```go
import (
    "math/rand"
    "time"
)

func randomNotIn(lst []int, n int) int {
    // Initialize the seed with the current time
    rand.Seed(time.Now().UnixNano())

    // Generate a number from 0 to n-1
    num := rand.Intn(n)

    // If the generated number is in the list, generate again
    for lst.Contains(num) {
        num = rand.Intn(n)
    }

    return num
}
```

File: dcp-solutions/2024-12-09/solution.c
```c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int uniformRandomNotInList(int n, int* l) {
    int total = 0;
    for (int i = 0; i <= n; i++) {
        total += 1;
    }

    while (1) {
        int num = rand() % total;

        int count = 0;
        for (int i = 0; i < n; i++) {
            if (l[i] == num)
                count++;
        }
        count++;

        if (count != 1) {
            return num;
        }
    }
}
```